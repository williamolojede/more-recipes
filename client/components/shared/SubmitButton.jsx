import React from 'react';
import ReactLoading from 'react-loading';
import PropTypes from 'prop-types';

const SubmitButton = ({ submitText, handleClick, submitLoading }) => (
  <button
    type="submit"
    className={`
    submit-btn z-depth-1
    ${submitLoading ? 'submit-loading' : ''}
    `}
    onClick={handleClick}
    disabled={submitLoading}
  >
    <span>{submitText}</span>
    <ReactLoading
      type="spokes"
      color="#fff"
      height="25px"
      width="25px"
      className="loading-icon"
    />

  </button>
);

SubmitButton.defaultProps = {
  handleClick() {}
};

SubmitButton.propTypes = {
  submitText: PropTypes.string.isRequired,
  handleClick: PropTypes.func,
  submitLoading: PropTypes.bool.isRequired
};

export default SubmitButton;
