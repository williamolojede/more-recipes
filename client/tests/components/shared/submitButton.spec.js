import React from 'react';
import SubmitButton from '../../../components/shared/SubmitButton';

function setup() {
  const props = {
    submitText: 'Send Message',
    submitLoading: false
  };
  const wrapper = shallow(<SubmitButton {...props} />);

  return {
    props,
    wrapper
  };
}


describe('Submit Button', () => {
  it('should render with right amount of elements', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should add a class name of submit-loading to button if submitLoading is true', () => {
    const { wrapper } = setup();
    wrapper.setProps({ submitLoading: true });
    expect(wrapper.find('button').hasClass('submit-loading')).toBe(true);
  });
});
