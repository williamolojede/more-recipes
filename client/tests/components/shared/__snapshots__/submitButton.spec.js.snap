// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submit Button should render with right amount of elements 1`] = `
<button
  className="
    submit-btn z-depth-1
    
    "
  disabled={false}
  onClick={[Function]}
  type="submit"
>
  <span>
    Send Message
  </span>
  <Loading
    className="loading-icon"
    color="#fff"
    delay={1000}
    height="25px"
    type="spokes"
    width="25px"
  />
</button>
`;
